packages/wp-dashboard/src/components/editorSettings/customFonts/index.js

```diff
@@ -187,47 +187,57 @@ function CustomFontsSettings({
   const handleOnSave = useCallback(async () => {
     if (canSave) {
       const urlWithProtocol = withProtocol(fontUrl);
-      let canFetch = false;
       try {
         await fetch(urlWithProtocol, {
           method: 'HEAD',
         });
-        canFetch = true;
       } catch (err) {
         setInputError(
           __(
             'Please ensure correct CORS settings for allowing font usage on this site.',
             'web-stories'
           )
         );
+        return;
       }
-      if (canFetch) {
-        try {
-          trackEvent('add_custom_font', {
-            url: urlWithProtocol,
-          });
-          const fontData = await getFontDataFromUrl(urlWithProtocol);
-          if (!fontData.family) {
-            setInputError(
-              __('Something went wrong, please try again.', 'web-stories')
-            );
-          } else {
-            await addCustomFont({ ...fontData, url: urlWithProtocol });
-            await fetchCustomFonts();
-            setFontUrl('');
-          }
-        } catch (err) {
+      let fontData;
+      try {
+        trackEvent('add_custom_font', {
+          url: urlWithProtocol,
+        });
+        fontData = await getFontDataFromUrl(urlWithProtocol);
+        if (!fontData.family) {
           setInputError(
-            sprintf(
-              /* translators: %s: list of allowed font types. */
-              __(
-                'Getting font data failed, please ensure the URL points directly to a %s file.',
-                'web-stories'
-              ),
-              translateToExclusiveList(ALLOWED_FONT_TYPES)
-            )
+            __('Something went wrong, please try again.', 'web-stories')
           );
+          return;
         }
+      } catch (err) {
+        setInputError(
+          sprintf(
+            /* translators: %s: list of allowed font types. */
+            __(
+              'Getting font data failed, please ensure the URL points directly to a %s file.',
+              'web-stories'
+            ),
+            translateToExclusiveList(ALLOWED_FONT_TYPES)
+          )
+        );
+        return;
+      }
+
+      try {
+        await addCustomFont({ ...fontData, url: urlWithProtocol });
+        await fetchCustomFonts();
+        setFontUrl('');
+      } catch (err) {
```

Comment:
unused `err` here

Reply:
Let's keep it, but add some `trackError('add_custom_font', err?.message)` while we're at it.

Reply:
Fixed.